sdk-core/src/test/java/dev/restate/sdk/core/StateTestSuite.java

```diff
@@ -76,6 +79,28 @@ public Stream<TestDefinitions.TestDefinition> definitions() {
                 getStateMessage("STATE"),
                 outputMessage(GreetingResponse.newBuilder().setMessage("Hello Francesco")))
             .named("Without GetStateEntry and completed with later CompletionFrame"),
+        testInvocation(this::getState, GreeterGrpc.getGreetMethod())
+            .withInput(
+                startMessage(2),
+                inputMessage(GreetingRequest.newBuilder().setName("Till")),
+                getStateMessage("STATE", new TerminalException(TerminalException.Code.CANCELLED)))
+            .onlyUnbuffered()
```

Comment:
This should also work with the buffered test executor (the one mimicking lambda)